src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -69,24 +73,54 @@
  * @author Timur Tibeyev.
  */
 public class XpathQueryGenerator {
+    /** Default distance between tab stops. */
+    private static final int DEFAULT_TAB_WIDTH = 4;
     /** The root ast. */
     private final DetailAST rootAst;
     /** The line number of the element for which the query should be generated. */
     private final int lineNumber;
     /** The column number of the element for which the query should be generated. */
     private final int columnNumber;
+    /** The file name. */
+    private final FileText fileText;
+    /** The distance between tab stop position. */
+    private final int tabWidth;
 
     /**
      * Creates a new {@code XpathQueryGenerator} instance.
      *
      * @param rootAst root ast
      * @param lineNumber line number of the element for which the query should be generated
      * @param columnNumber column number of the element for which the query should be generated
+     * @param fileName path to the file
+     * @param tabWidth distance between tab stop position
      */
-    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber) {
+    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber,
+                               String fileName, int tabWidth) {
         this.rootAst = rootAst;
         this.lineNumber = lineNumber;
         this.columnNumber = columnNumber;
+        try {
+            fileText = new FileText(new File(fileName),
+                    System.getProperty("file.encoding", "UTF-8"));
+        }
+        catch (IOException ex) {
+            throw new IllegalStateException("IOException occurred", ex);
+        }
+        this.tabWidth = tabWidth;
+    }
+
+    /**
+     * Creates a new {@code XpathQueryGenerator} instance.
+     *
+     * @param rootAst root ast
+     * @param lineNumber line number of the element for which the query should be generated
+     * @param columnNumber column number of the element for which the query should be generated
+     * @param fileName path to the file
+     */
+    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber,
+                               String fileName) {
+        this(rootAst, lineNumber, columnNumber, fileName, DEFAULT_TAB_WIDTH);
```

Comment:
@MEZk Do we really need a constructor without the `tabWidth`?

Reply:
@rnveach 
What should we do if user does not specify tabWidth? We need to check this outside the class and pass some default value to the constructor. The default value will be stored outside the class. I do not think that it is a good idea, if the class from the outside knows the default value.

Reply:
@MEZk If this generator will be called from `Main`, than `Main` should give a default value for tab width. I don't expect you would make 2 different constructor calls based on if tab width was supplied or not.
I expect call to be something similar to:
https://github.com/checkstyle/checkstyle/blob/2a5654734a64c5459819c5ced97bdacd0285ccc7/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L496-L506

Reply:
OK

@timurt 
Please, do

Reply:
done